spec/contracts/base_contract_spec.rb

```diff
@@ -0,0 +1,261 @@
+        end
+      end
+
+      it "returns it when not redefined in the child class" do
+        child_contract_class = Class.new(parent_contract_class)
+        contract = child_contract_class.new(model, user)
+        expect(contract.writable_attributes).to contain_exactly("name", "name_id")
+      end
+
+      it_behaves_like "the parent writable parameter is overridden by the child writable parameter"
+    end
+
+    context "when the attribute is defined as `writable: false` in the parent" do
+      let(:parent_contract_class) do
+        Class.new(BaseContract) do
+          attribute :name, writable: false
+        end
+      end
+
+      it "does not return it when not redefined in the child class" do
+        child_contract_class = Class.new(parent_contract_class)
+        contract = child_contract_class.new(model, user)
+        expect(contract.writable_attributes).to be_empty
+      end
+
+      it_behaves_like "the parent writable parameter is overridden by the child writable parameter"
+    end
+
+    context "when the attribute has not defined `:writable` in the parent" do
+      let(:parent_contract_class) do
+        Class.new(BaseContract) do
+          attribute :name
+        end
+      end
+
+      it "returns it when not redefined in the child class" do
+        child_contract_class = Class.new(parent_contract_class)
+        contract = child_contract_class.new(model, user)
+        expect(contract.writable_attributes).to contain_exactly("name", "name_id")
+      end
+
+      it_behaves_like "the parent writable parameter is overridden by the child writable parameter"
+    end
+
+    context "when the attribute is defined as `writable: -> { true }` in the parent" do
+      let(:parent_contract_class) do
+        Class.new(BaseContract) do
+          attribute :name, writable: -> { true }
+        end
+      end
+
+      it "returns it when not redefined in the child class" do
+        child_contract_class = Class.new(parent_contract_class)
+        contract = child_contract_class.new(model, user)
+        expect(contract.writable_attributes).to contain_exactly("name", "name_id")
+      end
+
+      it_behaves_like "the parent writable parameter is overridden by the child writable parameter"
+    end
+
+    context "when the attribute is defined as `writable: -> { false }` in the parent" do
+      let(:parent_contract_class) do
+        Class.new(BaseContract) do
+          attribute :name, writable: -> { false }
+        end
+      end
+
+      it "does not return it when not redefined in the child class" do
+        child_contract_class = Class.new(parent_contract_class)
+        contract = child_contract_class.new(model, user)
+        expect(contract.writable_attributes).to be_empty
+      end
+
+      it_behaves_like "the parent writable parameter is overridden by the child writable parameter"
+    end
+  end
+
+  describe "#reduce_by_writable_conditions"
+  describe "#reduce_writable_attributes"
```

Comment:
Remove these? or add a pending?

Reply:
yes, I should remove them. Good point. I thought I would test them, but it's not needed.

Reply:
Done